bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceAction.java

```diff
@@ -420,12 +419,14 @@ private void showOverlayInEditor() {
 				shellToUse = fShell;
 			}
 			overlay = new FindReplaceOverlay(shellToUse, fWorkbenchPart, fTarget);
+			hookDialogPreferenceListener();
 
 			FindReplaceOverlayFirstTimePopup.displayPopupIfNotAlreadyShown(shellToUse);
 		}
 
 		overlay.setPositionToTop(shouldPositionOverlayOnTop());
 		overlay.open();
+		overlay.getShell().addDisposeListener(__ -> removeDialogPreferenceListener());
```

Comment:
Wouldn't be better to move `overlayDialogPreferenceListener` to the `FindReplaceOverlay` itself? There one could simply add listener in constructor and remove on close().

Reply:
I had moved this listener out of the `FindReplaceOverlay` as a response to @HeikoKlare's comment:  https://github.com/eclipse-platform/eclipse.platform.ui/pull/1192#discussion_r1562438082

I believe that he is right and believe that it is indeed best if the Overlay is independent from the "global parametrization".